/*******************************************************************************
* Created  				Jun 23, 2021
* Modified				Jun 23, 2021
* Version				1.0.0.0

* FILE:					CloudeinAPI.proto
* Author:				dhimm@mslm.kr
* DESCRIPTION:			Cloudein TX-SM API grpc proto
* 
*******************************************************************************/

syntax = "proto3";

package cloudeinapi;

service CloudeinSession {
  rpc DoAllocateAPI (AllocateRequest) returns (AllocateResponse) {}
  rpc DoPrepareAPI (PrepareRequest) returns (PrepareResponse) {}
  rpc DoConnectAPI (ConnectRequest) returns (ConnectResponse) {}
  rpc DoReleaseAPI (ReleaseRequest) returns (ReleaseResponse) {}
}

/*
 *     Allocate API Request/Response message struct
*/
message AllocateRequest {
  string region = 1;
  string tx_name = 2;
}
message AllocateResponse {
  string region = 1;
  uint64 allocate_port1 = 11;
  uint64 allocate_port2 = 12;
  uint64 allocate_port3 = 13;
  uint64 allocate_port4 = 14;
  uint64 allocate_port5 = 15;
}

/*
 *     Prepare API Request/Response message struct
*/
message PrepareRequest {
  string region = 1;
  string tx_name = 2;
}
message PrepareResponse {
  string region = 1;
}

/*
 *     Connect API Request/Response message struct
*/
message ConnectRequest {
  string region = 1;
  string tx_name = 2;
}
message ConnectResponse {
  string region = 1;
}

/*
 *     Release API Request/Response message struct
*/
message ReleaseRequest {
  string region = 1;
  string tx_name = 2;
}
message ReleaseResponse {
  string region = 1;
}